File: tp1,1.cs
using System;
// Sandra Statuto

class Program
{
    // Enum de asteroides y metales.
    enum TamañoAsteroides { Pequeño, Mediano, Grande, Cataclísmico }
    enum TipoMetales { Hierro, Oro, Platino, MetalesMisceláneos }

    static void Main()
    {
        Random rand = new Random();
        int sistemaActual = 0;
        int totalHierro = 0, totalOro = 0, totalPlatino = 0, totalMetalesMisceláneos = 0, totalCarga = 0;

        while (true)
        {
            sistemaActual++;
            int asteroidesEnSistema = rand.Next(1, 11); // Generacion aleatoria de asteroides (entre 1 y 10).

            Console.WriteLine($"EN EL SISTEMA [{sistemaActual}] SE MINARON [{asteroidesEnSistema}] ASTEROIDES");

            for (int i = 0; i < asteroidesEnSistema; i++)
            {
                // Generador aleatorio para el asteroide.
                TamañoAsteroides tamaño = (TamañoAsteroides)rand.Next(0, 4);

                // Generador aleatoria para lacomposicion del asteroide.
                int[] composicion = GenerarComposicionAleatoria(tamaño, rand);

                Console.WriteLine($"{ObtenerNombreTamaño(tamaño)} asteroide:");
                ImprimirComposicion(composicion);

                // Actualizacion de la carga.
                totalHierro += composicion[(int)TipoMetales.Hierro];
                totalOro += composicion[(int)TipoMetales.Oro];
                totalPlatino += composicion[(int)TipoMetales.Platino];
                totalMetalesMisceláneos += composicion[(int)TipoMetales.MetalesMisceláneos];
                totalCarga += composicion[(int)TipoMetales.Hierro] +
                            composicion[(int)TipoMetales.Oro] +
                            composicion[(int)TipoMetales.Platino] +
                            composicion[(int)TipoMetales.MetalesMisceláneos];
            }

            Console.WriteLine($"Por un total de {totalCarga} KG de carga.");
            Console.WriteLine();

            Console.WriteLine("¿Desea entrar en otro sistema? (S para sí, cualquier otra tecla para salir del programa)");
            if (Console.ReadKey().Key != ConsoleKey.S)
            {
                break;
            }

            Console.WriteLine();
        }

        // Resumen de carga al salir del programa.
        Console.WriteLine("RESUMEN TOTAL:");
        Console.WriteLine($"Hierro total: {totalHierro} KG");
        Console.WriteLine($"Oro total: {totalOro} KG");
        Console.WriteLine($"Platino total: {totalPlatino} KG");
        Console.WriteLine($"Metales misceláneos total: {totalMetalesMisceláneos} KG");
        Console.WriteLine($"Carga total: {totalCarga} KG");
    }

    // Nombre del asteroide por tameño en función del valor enum.
    static string ObtenerNombreTamaño(TamañoAsteroides tamaño)
    {
        string[] nombres = { "Pequeño", "Mediano", "Grande", "Cataclísmico" };
        return nombres[(int)tamaño];
    }

    // Composición aleatoria para un asteroide en función de su tamaño.
    static int[] GenerarComposicionAleatoria(TamañoAsteroides tamaño, Random rand)
    {
        int[] composicion = new int[4];
        int totalCarga = 0;

        // Valores para los diferentes tamaños de asteroides.
        int[] valoresMin = { 100, 500, 1000, 2500 };
        int[] valoresMax = { 500, 1000, 2500, 5000 };

        // Valores aleatorios de metales.
        for (int i = 0; i < composicion.Length; i++)
        {
            composicion[i] = rand.Next(valoresMin[(int)tamaño], valoresMax[(int)tamaño] + 1);
            totalCarga += composicion[i];
        }

        // Asegurarse de que la carga total sea la especificada para el tamaño del asteroide.
        if (totalCarga != ObtenerCargaTamaño(tamaño))
        {
            // Ajustar de metales para que sume la carga correcta.
            double factorDeAjuste = (double)ObtenerCargaTamaño(tamaño) / totalCarga;
            for (int i = 0; i < composicion.Length; i++)
            {
                composicion[i] = (int)(composicion[i] * factorDeAjuste);
            }
        }

        return composicion;
    }

    // Carga total específica para un tamaño de asteroide.
    static int ObtenerCargaTamaño(TamañoAsteroides tamaño)
    {
        int[] cargas = { 1000, 2000, 5000, 10000 };
        return cargas[(int)tamaño];
    }

    // Salida por pantalla de la composición de un asteroide.
    static void ImprimirComposicion(int[] composicion)
    {
        string[] nombresMetales = { "Hierro", "Oro", "Platino", "Metales Misceláneos" };
        for (int i = 0; i < composicion.Length; i++)
        {
            Console.WriteLine($"{composicion[i]} KG de {nombresMetales[i]}");
        }
    }
}

File: TP1,1.py
import random

# Enumeraciones para representar tamaños de asteroides y tipos de metales.
class TamañoAsteroides:
    Pequeño, Mediano, Grande, Cataclísmico = range(4)

# Enumeración para tipos de metales.
class TipoMetales:
    Hierro, Oro, Platino, MetalesMisceláneos = range(4)

def obtener_nombre_tamaño(tamaño):
    nombres = ["Pequeño", "Mediano", "Grande", "Cataclísmico"]
    return nombres[tamaño]

def generar_composicion_aleatoria(tamaño, rand):
    composicion = [0, 0, 0, 0]
    total_carga = 0

    # Valores para los diferentes tamaños de asteroides.
    valores_min = [100, 500, 1000, 2500]
    valores_max = [500, 1000, 2500, 5000]

    # Valores aleatorios de metales.
    for i in range(len(composicion)):
        composicion[i] = rand.randint(valores_min[tamaño], valores_max[tamaño] + 1)
        total_carga += composicion[i]

    # Asegurarse de que la carga total sea la especificada para el tamaño del asteroide.
    if total_carga != obtener_carga_tamaño(tamaño):
        # Ajustar la composición para que sume la carga correcta.
        factor_de_ajuste = obtener_carga_tamaño(tamaño) / total_carga
        for i in range(len(composicion)):
            composicion[i] = int(composicion[i] * factor_de_ajuste)

    return composicion

def obtener_carga_tamaño(tamaño):
    cargas = [1000, 2000, 5000, 10000]
    return cargas[tamaño]

def imprimir_composicion(composicion):
    nombres_metal = ["Hierro", "Oro", "Platino", "Metales Misceláneos"]
    for i in range(len(composicion)):
        print(f"{composicion[i]} KG de {nombres_metal[i]}")

def main():
    rand = random.Random()
    sistema_actual = 0
    total_hierro = 0
    total_oro = 0
    total_platino = 0
    total_metales_miscelaneos = 0
    total_carga = 0

    while True:
        sistema_actual += 1
        asteroides_en_sistema = rand.randint(1, 10)  # Generar una cantidad aleatoria de asteroides (entre 1 y 10).

        print(f"EN EL SISTEMA [{sistema_actual}] SE MINARON [{asteroides_en_sistema}] ASTEROIDES")

        for i in range(asteroides_en_sistema):
            # Generar tamaño aleatorio para el asteroide.
            tamaño = rand.choice(list(TamañoAsteroides))
            composicion = generar_composicion_aleatoria(tamaño, rand)

            print(f"{obtener_nombre_tamaño(tamaño)} asteroide:")
            imprimir_composicion(composicion)

            # Actualizar la carga.
            total_hierro += composicion[TipoMetales.Hierro]
            total_oro += composicion[TipoMetales.Oro]
            total_platino += composicion[TipoMetales.Platino]
            total_metales_miscelaneos += composicion[TipoMetales.MetalesMisceláneos]
            total_carga += sum(composicion)

        print(f"Por un total de {total_carga} KG de carga.\n")

        respuesta = input("¿Desea entrar en otro sistema? (S para sí, cualquier otra tecla para salir del programa): ")
        if respuesta.lower() != 's':
            break

    print("RESUMEN TOTAL:")
    print(f"Hierro total: {total_hierro} KG")
    print(f"Oro total: {total_oro} KG")
    print(f"Platino total: {total_platino} KG")
    print(f"Metales misceláneos total: {total_metales_miscelaneos} KG")
    print(f"Carga total: {total_carga} KG")

if __name__ == "__main__":
    main()